<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>636</width>
    <height>361</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>聊天室客户端</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#titleLabel {
    background-color: rgba(99, 184, 255, 1);  /* 使用柔和的天蓝色 */
    color: white;  /* 白色字体以增强对比度 */
    border: none;
    border-bottom: 2px solid rgba(255, 255, 255, 0.5);  /* 使用半透明白色边框 */
    padding: 15px;  /* 增加内边距以改善可读性 */
    font: bold 24pt &quot;黑体&quot;;  /* 字体加粗 */
}

#mainFrame {
    border: none;
    background-color: rgba(240, 248, 255, 1);  /* 使用淡蓝色作为背景 */
}

#loginFrame {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #eef, stop: 1 #b2e1ff);  /* 使用明亮的渐变效果 */
    border: 1px solid #6bb8f8;  /* 用更柔和的蓝色边框 */
    padding: 20px;  /* 增加内边距，增强舒适度 */
    border-radius: 25px;
    box-shadow: 0 4px 15px rgba(0, 0, 0, 0.1);  /* 适度的阴影效果 */
}

/* 输入框 */
QLineEdit {
    border: 1px solid #ccc;  /* 使用浅灰色边框 */
    border-radius: 5px;
    padding: 10px;  /* 增加内边距 */
    background-color: rgba(255, 255, 255, 0.9); /* 明亮的白色背景 */
}

/* 按钮样式 */
QPushButton {
    background-color: rgba(66, 155, 255, 1);  /* 温和的蓝色 */
    color: white;
    border: none;
    border-radius: 5px;
    padding: 12px 24px;
    font-size: 16pt;
    cursor: pointer;  /* 显示手型光标 */
}

QPushButton:hover {
    background-color: rgba(38, 115, 255, 1);  /* 鼠标悬停时更深的蓝色 */
    cursor: pointer;  /* 确保鼠标悬停时是手型光标 */
}

QPushButton:pressed {
    background-color: rgba(30, 90, 255, 1);  /* 按下时更深的蓝色 */
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="titleLabel">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>多用户聊天室</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="chatPage">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QListWidget" name="userListWidget">
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="roomTextEdit"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLineEdit" name="sayLineEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="sayButton">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="logoutButton">
            <property name="text">
             <string>退出</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="loginPage">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>86</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>195</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <widget class="QFrame" name="loginFrame">
          <property name="frameShape">
           <enum>QFrame::Shape::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>服务器地址：</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="serverEdit">
               <property name="text">
                <string>127.0.0.1</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>昵称：</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="userNameEdit">
               <property name="text">
                <string>wiliu</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Policy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>10</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="loginButton">
             <property name="text">
              <string>登录聊天室</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>195</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="1">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>85</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
